pkgs/applications/radio/wfview/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv, lib, fetchFromGitLab, qcustomplot, qt5, libopus, eigen, portaudio
+, rtaudio, hidapi, udev, pulseaudio }:
+stdenv.mkDerivation rec {
+  pname = "wfview";
+  version = "1.64";
+  src = fetchFromGitLab {
+    owner = "eliggett";
+    repo = "wfview";
+    rev = "v${version}";
+    sha256 = "QdrCji+gktrWfJKf9+yPkmdzfXOMIh4lwG2JA8WVUT8=";
+  };
+  patches = [ ./remove-syscalls.patch ];
```

Comment:
Why?

Reply:
This was a fun one.

The upstream build script uses system calls to gather information about the system to bake into the build for informational purposes (calls directly to hostname, whoami, and git to grab the version). The first two violate hermeticity, and the git call assumes the build is being performed from within a git repository rather than an archive tarball.

The upstream build script also invokes /sbin/ldconfig directly to search for qcustomplot to determine how to link it, due to varying capitalisation options. In the Nix build, we already have qcustomplot on the link path and know how to link it.

The patch resolves this by removing those calls and replacing them with placeholders.

But it does highlight that a comment would be handy to explain why it's there. Additionally, I'll see if I can inject the git version in from the derivation rather than using a placeholder.